### ISSUE TITLE
Our code has an incompatiblity with updated versions of @octokit/rest from 18.1.0 to 18.3.0

### ISSUE THREAD

**Issue by dannyvv** [microsoft/react-native-windows#7239]
We need to fix our code so we can take updates for: @octokit/rest from 18.1.0 to 18.3.0
The error we have is:
```
 [6:36:30 PM] x src/createGithubReleases.ts(95,5): error TS2769: No overload matches this call.
ERR!   The last overload gave the following error.
ERR!     Argument of type '{ (params?: (RequestParameters & Pick<{ owner: string; repo: string; } & { per_page?: number | undefined; page?: number | undefined; }, "per_page" | "page" | "owner" | "repo">) | undefined): Promise<...>; defaults: <O extends RequestParameters = RequestParameters>(newDefaults: O) => RequestInterface<...>; endpoint: ...' is not assignable to parameter of type 'RequestInterface<object>'.
ERR!       The types of 'endpoint.DEFAULTS' are incompatible between these types.
ERR!         Type '{ url: string; } & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string | undefined; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type 'object & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string | undefined; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }'.
ERR!           Type '{ url: string; } & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string | undefined; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type 'RequestParameters'.
ERR!             Types of property 'request' are incompatible.
ERR!               Type 'import("D:/a/_work/1/s/node_modules/@octokit/plugin-rest-endpoint-methods/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions | undefined' is not assignable to type 'import("D:/a/_work/1/s/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions | undefined'.
ERR!                 Type 'import("D:/a/_work/1/s/node_modules/@octokit/plugin-rest-endpoint-methods/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions' is not assignable to type 'import("D:/a/_work/1/s/node_modules/@octokit/types/dist-types/RequestRequestOptions").RequestRequestOptions'.
ERR!                   Types of property 'agent' are incompatible.
ERR!                     Type 'unknown' is not assignable to type 'Agent | undefined'.
ERR!                       Type 'unknown' is not assignable to type 'Agent'.
ERR! src/createGithubReleases.ts(100,42): error TS2571: Object is of type 'unknown'.
ERR! src/createGithubReleases.ts(106,44): error TS2345: Argument of type 'PaginationResults<unknown>' is not assignable to parameter of type '{ tag_name: string; }[]'.
ERR!   Type 'unknown' is not assignable to type '{ tag_name: string; }'.
```

For now updates to this pacakge have been postponed. See #7236

**Comment by chrisglein** [comment]
Blocked on updating to latest but doesn't appear to be a functional regression yet (typescript is confused).
